FHEM/10_BindingsIo.pm

```diff
@@ -18,64 +17,77 @@ use CoProcess;
 use JSON;
 use Time::HiRes qw(time);
 
-sub Log($$);
-sub Log3($$$);
+our $init_done;
+our $modules;
+
 
 my $USE_DEVIO_DECODEWS = 0;
 my $timeouts = 0;
 
-sub
-BindingsIo_Initialize($)
-{
+sub BindingsIo_Initialize {
   my ($hash) = @_;
 
   $hash->{parseParams} = 1;
 
-  $hash->{DefFn}    = 'BindingsIo_Define';
-  $hash->{UndefFn}  = 'BindingsIo_Undefine';
-  $hash->{GetFn}    = 'BindingsIo_Get';
-  $hash->{SetFn}    = 'BindingsIo_Set';
-  $hash->{AttrFn}   = 'BindingsIo_Attr';
+  $hash->{DefFn}    = \&BindingsIo_Define;
+  $hash->{UndefFn}  = \&BindingsIo_Undefine;
+  $hash->{GetFn}    = \&BindingsIo_Get;
+  $hash->{SetFn}    = \&BindingsIo_Set;
+  $hash->{AttrFn}   = \&BindingsIo_Attr;
   $hash->{AttrList} = $readingFnAttributes;
-  $hash->{NotifyFn}   = 'BindingsIo_Notify';
+  $hash->{NotifyFn}   = \&BindingsIo_Notify;
 
-  $hash->{ReadFn}   = 'BindingsIo_Read';
-  $hash->{ReadyFn}  = 'BindingsIo_Ready';
-  $hash->{WriteFn}  = 'BindingsIo_Write';
+  $hash->{ReadFn}   = \&BindingsIo_Read;
+  $hash->{ReadyFn}  = \&BindingsIo_Ready;
+  $hash->{WriteFn}  = \&BindingsIo_Write;
 
   $hash->{Clients} = "PythonModule:fhempy"; # NodeModule
 
-  return undef;
+  return ;
 }
 
-sub
-BindingsIo_Define($$$)
-{
-  my ($hash, $a, $h) = @_;
+sub BindingsIo_Define {
+  my ( $hash, $a ) = @_;
```

Comment:
Aus welchem Grund ist $h entfernt worden? Wenn `parseParams` aktiv ist, werden die Parameter in Array und Hash aufgeteilt, oder habe ich etwas übersehen?

Reply:
Laut deiner commandref gibt es keine key=Value Parameter, die define akzeptiert.
Daher braucht es auch kein Assignment einer Variable die nichts enthalten soll.

Vielleicht habe ich aber auch etwas übersehen?